C/sort/mergesort_in_O(1)_space_complexity.c

```diff
@@ -0,0 +1,114 @@
+// algorithm for merging two sorted array in O(1) space complexity and in
+// Nlog(N) + Mlog(M) time complexity
+#include <stdio.h>
+#include <stdlib.h>
+int i, j;
+
+void mergesort(int a[], int b[], int n, int m) {
+  int min;
+  if (n >= m) {
+    min = m;
+  } else {
+    min = n;
+  }
+  for (i = 0; i < min; i++)  // we are swapping the elements of both the array
+    // which should be there in the result
+  {
+    if (a[n - i - 1] > b[i]) {
+      a[n - i - 1] = a[n - i - 1] - b[i];  // swapping the elements
+      b[i] = a[n - i - 1] + b[i];
```

Comment:
@ag278 I think you are making both the array bitonic i.e. making the sequence of numbers which is first strictly increasing then after a point strictly decreasing so you can mention this in comments.

Reply:
@iamrajiv I am not making the arry bitonic. I am just transferring the elements fron the arrays in such a way that after swapping the elements of array both the arrays should contain only those elements which would be present in the final arrays after sorting the arrays. I am using similar type of  inplace algorithm for that